include/rotatingMHD/solver_class.h

```diff
@@ -344,19 +432,45 @@ LinearAlgebra::MPI::Vector        projection_step_rhs;
  */
 double                            norm_projection_step_rhs;
 
+/*!
+ * @brief The system matrix of the zeroth step.
+ *
+ * @details The system matrix corresponds to the stiffness matrix of the
+ * Lagrange multiplier field. Its entries only change if the
+ * spatial discretization changes. It is stored in memory because
+ * otherwise an assembly would be required if the timestep changes.
+ *
+ */
+LinearAlgebra::MPI::SparseMatrix  zeroth_step_system_matrix;
+
+/*!
+ * @brief Vector representing the right-hand side of the linear system
+ * of the zeroth step performed to compute admissible initial conditions
+ * for the Lagrange multiplier.
+ *
+ */
+LinearAlgebra::MPI::Vector        zeroth_step_rhs;
+
 /*!
  * @brief The preconditioner of the diffusion step.
  *
  */
 std::shared_ptr<LinearAlgebra::PreconditionBase> diffusion_step_preconditioner;
 
-
 /*!
  * @brief The preconditioner of the projection step
  *
  */
 std::shared_ptr<LinearAlgebra::PreconditionBase> projection_step_preconditioner;
 
+/*!
+ * @brief The preconditioner of the pre-step performed to compute admissible initial conditions
+ * for the Lagrange multiplier
+ *
+ */
+std::shared_ptr<LinearAlgebra::PreconditionBase> zeroth_step_preconditioner;
```

Comment:
Same here regarding the naming of the variables. This preconditioner is only used and could constructed (and destroyed) inside the `solve_poisson_pre_step` method.